libdebug/debugger/internal_debugger.py

```diff
@@ -408,26 +430,13 @@ def breakpoint(
             address = self.resolve_address(position, file)
             position = hex(address)
 
-        if condition:
-            if not hardware:
-                raise ValueError(
-                    "Breakpoint condition is supported only for hardware watchpoints.",
-                )
-
-            if condition.lower() not in ["w", "rw", "x"]:
-                raise ValueError(
-                    "Invalid condition for watchpoints. Supported conditions are 'w', 'rw', 'x'.",
-                )
+        if condition != "x" and not hardware:
+            raise ValueError("Breakpoint condition is supported only for hardware watchpoints.")
 
-            if length not in [1, 2, 4, 8]:
-                raise ValueError(
-                    "Invalid length for watchpoints. Supported lengths are 1, 2, 4, 8.",
-                )
-
-        if hardware and not condition:
-            condition = "x"
+        bp = Breakpoint(address, position, 0, hardware, callback, condition.lower(), length)
```

Comment:
here condition is already forced to be lower case (see comments in breakpoint_validator)